soc/arm/st_stm32/common/soc_config.c

```diff
@@ -30,6 +32,32 @@ static int st_stm32_common_config(const struct device *dev)
 	DBGMCU->CR |= DBGMCU_CR_TRACE_IOEN;
 #endif
 
+#if defined(CONFIG_USE_SEGGER_RTT) && defined(__HAL_RCC_DMA1_CLK_ENABLE)
+	/* On some STM32 boards, for unclear reason,
+	 * RTT feature is working with realtime update only when
+	 *   - one of the DMA is clocked.
+	 * See https://github.com/zephyrproject-rtos/zephyr/issues/34324
+	 */
+	__HAL_RCC_DMA1_CLK_ENABLE();
+#endif
+
+	/* On some STM32 boards, for unclear reason,
+	 * RTT feature is working with realtime update only when
+	 *   - one of the DBGMCU bit STOP/STANDBY/SLEEP is set
+	 * See https://github.com/zephyrproject-rtos/zephyr/issues/34324
+	 */
+#if defined(LL_APB1_GRP1_PERIPH_DBGMCU)
+	LL_APB1_GRP1_EnableClock(LL_APB1_GRP1_PERIPH_DBGMCU);
+#elif defined(LL_APB1_GRP2_PERIPH_DBGMCU)
+	LL_APB1_GRP2_EnableClock(LL_APB1_GRP2_PERIPH_DBGMCU);
+#elif defined(LL_APB2_GRP1_PERIPH_DBGMCU)
+	LL_APB2_GRP1_EnableClock(LL_APB2_GRP1_PERIPH_DBGMCU);
+#endif
+#if defined(CONFIG_SOC_SERIES_STM32H7X) || defined(CONFIG_SOC_SERIES_STM32MP1X)
+	HAL_EnableDBGSleepMode();
+#else
+	LL_DBGMCU_EnableDBGStopMode();
+#endif
```

Comment:
Should this be also under `#ifdef CONFIG_USE_SEGGER_RTT` ?

Reply:
Yes. Done